gb.go

```diff
@@ -55,13 +55,28 @@ func copyfile(dst, src string) error {
 	return err
 }
 
+// RunErr is returned when an exec.Command fails
+type RunErr struct {
+	Err    error
+	Cmd    string
+	Args   []string
+	Output string
+}
+
+func (e RunErr) Error() string {
+	s := fmt.Sprintf("Run Error: %v\n", e.Err)
+	s += fmt.Sprintf("# %s %s\n", e.Cmd, strings.Join(e.Args, " "))
+	s += e.Output
+	return s
+}
+
 func run(dir, command string, args ...string) error {
 	var buf bytes.Buffer
 	err := runOut(&buf, dir, command, args...)
 	if err != nil {
-		fmt.Printf("# %s %s\n%s", command, strings.Join(args, " "), buf.String())
+		return RunErr{Err: err, Cmd: command, Args: args, Output: buf.String()}
```

Comment:
why can't this just be

err = fmt.Errorf("# %s %s: %v\n%s", command, strings.Join(args, " "), err, buf.String())